python/ql/src/external/ExternalArtifact.qll

```diff
@@ -1,3 +1,7 @@
+/**
+ * Provides classes for working with external data.
+ */
```

Comment:
Having never heard about this before, I have no idea what this is actually about by just reading this comment. Looking through the source code, I can see this is used by the Thrift code. So I guess `external data` in this context means data that is part of the DB that is not Python code? (or would something like CSV and XML not be a part of this?)

Maybe we can expand this comment, or the one for `ExternalData`? :blush:

Reply:
To be quite honest, I _don't_ know exactly what these classes are used for. That's why I left some of the undocumented. (I can certainly make an educated guess about what they might represent, but I wouldn't feel comfortable making this the official documentation.)

As for the level of documentation, I can only refer you to the JavaScript equivalent: https://github.com/github/codeql/blob/master/javascript/ql/src/external/ExternalArtifact.qll
:wink: